src/routes/games/snake/+page.svelte

```diff
@@ -1,182 +1,211 @@
-      background-color: pink;
-    }
-    .snake {
-      background-color: green;
-    }
-    .row {
-      display: flex;
-    }
-    .center {
-      display: flex;
-      justify-content: center;
-      align-items: center;
-    }
-    .tcenter {
-      text-align: center;
-    }
-    .restart {
-      margin-top: 10px;
-    }
-  
-    .enter {
-      border: #000 1px solid;
-      background-color: #eee;
-      border-radius: 2px;
-      padding: 5px;
-    }
-  </style>
-  
-  <svelte:window
-    on:keydown={(e) => {
-      switch (e.key) {
-        case 'ArrowLeft':
-          direction = [0, -1];
-          break;
-        case 'ArrowRight':
-          direction = [0, 1];
-          break;
-        case 'ArrowUp':
-          direction = [-1, 0];
-          break;
-        case 'ArrowDown':
-          direction = [1, 0];
-          break;
-        case 'Enter':
-          restart();
-          break;
-      }
-    }} />
-  
-  <main>
-    {#if lost}
-      <h1 class="tcenter">you lost</h1>
-      <h3 class="tcenter">Hit <code class="enter">ENTER</code> to restart</h3>
-    {/if}
-    <h3 class="tcenter">snake length {snakePosition.length}</h3>
-    <div class="center">
-      <div>
-        {#each gridWithSnake as row, i}
-          <div class="row">
-            {#each row as cell, k}
-              <div
-                on:click={() => (grid[i][k] = 'food')}
-                class={`square ${cell}`} />
-            {/each}
-          </div>
-        {/each}
-      </div>
-    </div>
-  
-    {#if lost}
-      <div class="center restart">
-        <button on:click={restart}> Start again </button>
-      </div>
-    {/if}
-  </main>
\ No newline at end of file
+	import { PUBLIC_API_URL } from '$env/static/public';
+	import { onMount } from 'svelte/internal';
+
+	const TICK_DELAY = 200;
+	const GRID_SIZE = 20;
+	let SNAKE_HEAD = 0;
+	let lost = false;
+	type Cell = 'empty' | 'snake' | 'food';
+	let grid: Cell[][] = [...Array(GRID_SIZE)].map(() => [...Array(GRID_SIZE)].map(() => 'empty'));
+	let snakePosition: Array<[number, number]> = [[12, 13]];
+	let direction = [0, 1];
+	let gridWithSnake = grid;
+	export let data;
+	let highscore: number = 0;
+
+	function getRandomInt(max: number) {
+		return Math.floor(Math.random() * Math.floor(max));
+	}
+
+	function randomFood() {
+		while (true) {
+			let x_food: number = getRandomInt(GRID_SIZE);
+			let y_food: number = getRandomInt(GRID_SIZE);
+			if (gridWithSnake[x_food][y_food] != 'snake') {
+				grid[x_food][y_food] = 'food';
+				break;
+			}
+		}
+	}
+
+	async function uploadHighscore() {
+		try {
+			const response = await fetch('/api/snake', {
+				method: 'POST',
+				headers: { 'Content-Type': 'application/json' },
+				body: JSON.stringify({ score: snakePosition.length })
+			});
+		} catch (err) {
+			console.log(err);
+		}
+	}
+
+	export async function getHighscore() {
+		const url = PUBLIC_API_URL + '/snake/user?userId=' + data.user.id;
+
+		const response = await fetch(url);
+
```

Comment:
du musst hier nicht nochmal extra die highscores fetchen, du hast die ja schon in der page.server.ts im Pageserverload geladen